_posts/2024-06-22-custom-copr-builders.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+title: "Need something extra? Employ your own Copr builders!"
+date: 2024-06-22 00:00:00 +0000
+lang: en
+---
+
+In Copr, we deploy the [Resalloc server][resalloc] to manage our farm of
+builders.  The _Copr_ build system (specifically the _copr-backend_ package)
+acts as the "client," creating tickets with _Resalloc_ and requesting specific
+kinds of worker virtual machines ("resources") for particular build tasks.
+To do this, _Copr_ uses a known set of per-ticket tags to describe particular
+needs (e.g., `--tag arch_x86_64`, `--tag powerful`, and so on).
+
+_Resalloc's_ task is to match the _client's_ ticket tags with pre-allocated VMs
+(or allocate new VMs if needed).
+
+A [recent contribution][per-pkg-tags] has enabled Copr administrators to request
+high-performance builders for specific build tasks.  And as a nice side effect,
+it also opened the door to providing any other **custom workers**.  More
+importantly for you, if you don't want to wait in the queue anymore, you can now
+have **workers dedicated to your team**!
+
+
+What's Needed?
+--------------
+
+Copr administrators should be able to configure a specific _Resalloc_ pool of
+VMs (builders) for you and make *your builds* use them.  Of course, they'll need
+your permission to start the machines in your name (_IBM Cloud_ or _EC2_ token
+needed, SSH key granting access to your VM hypervisor machine, or
+similar).
```

Comment:
Can the custom builders be anywhere or does it have to be something that resalloc can interact with?
https://github.com/praiskup/resalloc?tab=readme-ov-file#what-do-you-mean-by-resources
Either way, I'd be more explicit about it.

Reply:
I dont think we have to limit this, and good idea with making it explicit 👌

E.g. using Azure would be more work for us (golden image preparation, scripting, debugging cloud issues from time to time..). And I'm curious if we (the team) want that at all (i'd vote for yes, just to learn something new, but..).

Reply:
Updated.